src/entities/theme.rs

```diff
@@ -0,0 +1,26 @@
+extern crate sdl2;
+
+use self::sdl2::pixels::Color;
+use self::sdl2::rwops::RWops;
+use self::sdl2::ttf::Font;
+use self::sdl2::ttf::Sdl2TtfContext;
+
+
+pub struct Theme<'a> {
+    pub color: Color,
+    pub font: Font<'a, 'a>,
+    pub font_size: u16,
+}
+
+impl<'a> Theme<'a> {
+    pub fn new(color: Color, font_bytes: &'a[u8], font_size: u16, ttf_context: &'a Sdl2TtfContext) -> Theme<'a> {
+        let ttf_rwops = RWops::from_bytes(font_bytes).unwrap();
+        let font = ttf_context.load_font_from_rwops(ttf_rwops, font_size).unwrap();
```

Comment:
This is the part where it falls apart. The `font` object keeps a borrowed reference to `ttf_context` and even though it has the same lifetime as the `Theme` object, it still fails the compiler lifetime checks.

Reply:
```
cannot return value referencing local variable `ttf_context`

returns a value referencing data owned by the current functionrustc(E0515)
game.rs(106, 13): `ttf_context` is borrowed here
game.rs(109, 9): returns a value referencing data owned by the current function
```

Reply:
I think we're good now.